rocAL/rocAL_pybind/amd/rali/decoders.py

```diff
@@ -0,0 +1,211 @@
+import amd.rali.types as types
+import rali_pybind as b
+from amd.rali.pipeline import Pipeline
+
+def image(*inputs, user_feature_key_map = None, path='', file_root ='', annotations_file= '', shard_id = 0, num_shards = 1, random_shuffle = False, affine=True, bytes_per_sample_hint=0, cache_batch_copy= True, cache_debug = False, cache_size = 0, cache_threshold = 0,
+                 cache_type='', device_memory_padding=16777216, host_memory_padding=8388608, hybrid_huffman_threshold= 1000000, output_type = types.RGB,
+                 preserve=False, seed=-1, split_stages=False, use_chunk_allocator= False, use_fast_idct = False, device = None):
+    reader = Pipeline._current_pipeline._reader
+    if( reader == 'COCOReader'):
```

Comment:
As we discussed in a call for external_source support, we might need a decoder and reader separate for external source support. In next version we should have reader return batch of images and meta_data, and decoder needs to work on those input as well.

Reply:
Sure Rajy. In our last call we discussed we will introduce option to take file names or decoded image as input to rocAL. We will discuss the implementation details and add it in next version.